### PR TITLE
caf: Add LEDs module.

### PR DESCRIPTION
**Pull request by emob-nordic**
LEDs module for CAF.

- [x] https://github.com/nrfconnect/sdk-nrf/pull/4079 must be merged first.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

### CODE REVIEW

**Review by pdunaj — APPROVED** [review]

**Comment by emob-nordic** [comment]
This PR must wait before https://github.com/nrfconnect/sdk-nrf/pull/4079 is merged.

**Review by pdunaj — APPROVED** [review]

File: subsys/caf/modules/leds.c [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	/* If event is unhandled, unsubscribe. */
+	__ASSERT_NO_MSG(false);
+
+	return false;
+}
+
+EVENT_LISTENER(MODULE, event_handler);
+EVENT_SUBSCRIBE(MODULE, led_event);
+EVENT_SUBSCRIBE(MODULE, module_state_event);
+#if CAF_LEDS_PM_EVENTS
```
Comment (MarekPieta) [inline subsys/caf/modules/leds.c:270]: `CONFIG_CAF_LEDS_PM_EVENTS`

File: applications/nrf_desktop/src/modules/Kconfig.led_stream [hunk 2]
```diff
@@ -8,7 +8,7 @@ menu "LED stream"
 
 config DESKTOP_LED_STREAM_ENABLE
 	bool "Enable LED stream"
-	depends on DESKTOP_LED_ENABLE && DESKTOP_CONFIG_CHANNEL_ENABLE
+	depends on CAF_LED_EVENTS && DESKTOP_CONFIG_CHANNEL_ENABLE
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.led_stream:11]: CONFIG_CAF_LEDS, we actually need the module that controls the HW LEDS

File: subsys/caf/modules/leds.c (hunk 1 above)
Comment (emob-nordic) [reply subsys/caf/modules/leds.c:270]: whopsie.. Thanks!

File: applications/nrf_desktop/src/modules/Kconfig.led_stream (hunk 2 above)
Comment (emob-nordic) [reply applications/nrf_desktop/src/modules/Kconfig.led_stream:11]: Yes, but from SW perspective we need events and nothing more.

File: subsys/caf/modules/leds.c [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+		k_delayed_work_submit(&led->work, K_MSEC(next_delay));
+	} else {
+		LOG_WRN("LED effect with no effect");
+	}
+}
+
+static int leds_init(void)
+{
+	static const char * const dev_name[] = {
+#if defined(CONFIG_PWM) && DT_NODE_HAS_STATUS(DT_NODELABEL(pwm0), okay)
```
Comment (MarekPieta) [inline subsys/caf/modules/leds.c:131]: Maybe we could even assert that CONFIG_PWM is enabled (it's a Kconfig dependency of the module)

File: applications/nrf_desktop/src/modules/Kconfig.led_stream (hunk 2 above)
Comment (emob-nordic) [reply applications/nrf_desktop/src/modules/Kconfig.led_stream:11]: @pdunaj What is your statement? I remember we talked about that offline.

File: subsys/caf/modules/leds.c (hunk 3 above)
Comment (pdunaj) [reply subsys/caf/modules/leds.c:131]: Could add it but I don't think there is a need if there is a dependency in Kconfig.

File: applications/nrf_desktop/src/modules/Kconfig.led_stream (hunk 2 above)
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/Kconfig.led_stream:11]: @emob-nordic is right. In the end we don't care who will handle the request. Integrator of nrf_desktop can add proprietary module handling leds.

File: subsys/caf/modules/leds.c (hunk 3 above)
Comment (emob-nordic) [reply subsys/caf/modules/leds.c:131]: I leave this as is for now.

File: applications/nrf_desktop/src/modules/Kconfig.led_stream (hunk 2 above)
Comment (emob-nordic) [reply applications/nrf_desktop/src/modules/Kconfig.led_stream:11]: Left with `*_EVENTS`.

**Review by MarekPieta — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**